### PR TITLE
CXF-8725: Allow RetryStrategy to optionally not retry for 404 responses

### PR DESCRIPTION
**Pull request by kwin**
"org.apache.cxf.transport.service_not_available" for 404 responses

### CODE REVIEW

File: rt/transports/http/src/main/java/org/apache/cxf/transport/http/HTTPConduit.java [hunk 1]
```diff
@@ -1600,7 +1605,7 @@ protected int doProcessResponseCode() throws IOException {
             }
             if (exchange != null) {
                 exchange.put(Message.RESPONSE_CODE, rc);
-                if (rc == 404 || rc == 503 || rc == 429) {
+                if ((rc == 404 && MessageUtils.getContextualBoolean(outMessage, SERVICE_NOT_AVAILABLE_ON_HTTP_404, true)) || rc == 503 || rc == 429) {
```
Comment (reta) [inline rt/transports/http/src/main/java/org/apache/cxf/transport/http/HTTPConduit.java:1608]: Thanks @kwin , I am wondering if we could use list of status codes as contextual property instead (`404`, `503`, `429` being the default)?

Comment (kwin) [reply rt/transports/http/src/main/java/org/apache/cxf/transport/http/HTTPConduit.java:1608]: Makes sense, I am gonna change the PR

Comment (kwin) [reply rt/transports/http/src/main/java/org/apache/cxf/transport/http/HTTPConduit.java]: Done in https://github.com/apache/cxf/pull/964/commits/191bea69cff1ff282d6565d0ba1e4b49daa6d95b

**Review by reta — APPROVED** [review]
Thank you, @kwin !

**Pull request merged**